milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
@efischer19 if `relationship` is `None`, it will return all the milestones for the course. But when the relationship is specified, we only want the milestones of that relationship type, and if the type is unknown, then there are never milestones of that type.